### PR TITLE
Add infrastructure for versioning

### PR DESCRIPTION
**Pull request by elrayle**
* update README to include…
  * how to specify the gem name and version that a document describes
  * show examples of info, question, and warning boxes
* number pages for customizing metadata tutorial to make it easier to track page order
* update a-z index
* use what happens when I deposit and customizing metadata tutorial as examples for versioning

### CODE REVIEW

**Comment by elrayle** [comment]
This is a good start, but could still use some thought for how it will work nicely in the theme. 

#### Works Well
* It is easy to add gem's name and version to every file.
* It is easy to add selection list which switches between versions.

#### What doesn't work well
* version in the left hand menu
  * Option 1: Leave all versions in the menu tacking on the version info to the title in the side bar.  
    * CON: This has duplicate titles in the sidebar which will only get worse as the number of versions grows.  This will be magnified for multi-page tutorials (e.g. Customizing Metadata).
    * PRO: All version pages have a presence in the sidebar.
  * Option 2: Only put the latest version in the sidebar
    * PRO: Users are driven to latest version which is most likely what they want.
    * PRO: Other versions are easily accessible from the version selector.
    * CON: When an alternate version is selected, the menu fully retracts.  The user loses the context of where they are in the documentation.  This especially effects multi-page tutorials (e.g. Customizing Metadata).
    * CON: If only the latest version is available in the side menu, then every time you click another page in a multi-page tutorial, you will be back on the latest version and have to select the version you want every time.  (NOTE: This does not effect the NEXT/PREV buttons as they are coded to a specific versioned tutorial.)

#### Bottom line

I think is is ok to accept this PR as is, but we need to continue to think through how to represent pages that exist for multiple versions of a gem.

**Comment by elrayle** [comment]
Relates to...
Issue #37 Versioning of Documentation for Changes in New Releases
Issue #102 Create initial doc for "best practices" for writing documentation (so meta!)
